src/ilcdlib/entity/compliance.py

```diff
@@ -0,0 +1,117 @@
+#
+#  Copyright 2023 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+from typing import Type
+
+from openepd.model.standard import Standard
+
+from ilcdlib.common import (
+    BaseIlcdMediumSpecificReader,
+    IlcdXmlReader,
+    OpenEpdComplianceSupportReader,
+    OpenEpdStandardReader,
+)
+from ilcdlib.dto import IlcdReference
+from ilcdlib.type import LangDef
+from ilcdlib.utils import none_throws
+from ilcdlib.xml_parser import T_ET
+
+
+class IlcdStandardReader(OpenEpdStandardReader, IlcdXmlReader):
+    """Reader that can parser an ILCD Standard specification from an XML file."""
+
+    def __init__(
+        self,
+        element: T_ET.Element,
+        data_provider: BaseIlcdMediumSpecificReader,
+    ):
+        super().__init__(data_provider)
+        self._entity = element
+
+    def get_uuid(self) -> str:
+        """Get the UUID of the entity described by this data set."""
+        return none_throws(
+            self._get_text(self._entity, ("source:sourceInformation", "source:dataSetInformation", "common:UUID"))
+        )
+
+    def get_version(self) -> str | None:
+        """Get the version of the entity described by this data set."""
+        return self._get_text(
+            self._entity,
+            ("source:administrativeInformation", "source:publicationAndOwnership", "common:dataSetVersion"),
+        )
+
+    def get_own_reference(self) -> IlcdReference | None:
+        """Get the reference to this data set."""
+        return IlcdReference(entity_type="sources", entity_id=self.get_uuid(), entity_version=self.get_version())
+
+    def get_short_name(self) -> str:
+        """Get the short name of the entity described by this data set."""
+        short_name = self._get_text(
+            self._entity,
+            ("source:sourceInformation", "source:dataSetInformation", "common:shortName"),
+        )
+        return short_name or f"Name is missing. [Standard {self.get_uuid()}]"
```

Comment:
Is there a reason not to return `None`?

Reply:
A short name is required from the Standard OpenApi object. So the name should be populated here or on Standard creation. 

Or, we can skip such standards

Reply:
No need to follow validation rules in this context. We extract whatever we have.